stonesoup/updater/particle.py

```diff
@@ -221,3 +224,207 @@ def predict_measurement(self, state_prediction, *args, **kwargs):
+            hypothesis.prediction.particles)
+
+        return ParticleStateUpdate(None,
+                                   particle_list=new_particles,
+                                   hypothesis=hypothesis,
+                                   timestamp=hypothesis.measurement.timestamp)
+
+    @lru_cache()
+    def predict_measurement(self, state_prediction, measurement_model=None,
+                            **kwargs):
+
+        if measurement_model is None:
+            measurement_model = self.measurement_model
+
+        new_particles = []
+        for particle in state_prediction.particles:
+            new_state_vector = measurement_model.function(
+                particle.state_vector, noise=False, **kwargs)
+            new_particles.append(
+                RaoBlackwellisedParticle(new_state_vector,
+                                         weight=particle.weight,
+                                         parent=particle.parent,
+                                         model_probabilities=particle.model_probabilities))
+
+        return ParticleMeasurementPrediction(
+            None, particle_list=new_particles, timestamp=state_prediction.timestamp)
+
+    @staticmethod
+    def calculate_model_probabilities(particle, measurement_model, position_mapping,
+                                      transition_matrix, transition_models, time_interval):
+        """Calculates the new model probabilities based
+            on the ones calculated in the previous time step"""
+
+        previous_probabilities = particle.model_probabilities
+
+        denominator_components = []
+        # Loop over the current models m_k
+        for l in range(len(previous_probabilities)):  # noqa: E741
+            selected_model_sum = 0
+            # Loop over the previous models m_k-1
+            for i, model in enumerate(transition_models):
+                # Looks up p(m_k|m_k-1)
+                # Note: if p(m_k|m_k-1) = 0 then p(m_k|x_1:k) = 0
+                transition_probability = transition_matrix[i][l]
+                if transition_probability == 0:
+                    break
+                # Getting required states to apply the model to that state vector
+                parent_required_state_space = copy.copy(particle.parent)
+                parent_required_state_space.state_vector = \
+                    particle.parent.state_vector[position_mapping[i], :]
+
+                # The noiseless application of m_k onto x_k-1
+                mean = model.function(
+                    parent_required_state_space, time_interval=time_interval, noise=False)
+
+                # Input the indices that were removed previously
+                for j in range(len(particle.state_vector)):
+                    if j not in position_mapping[i]:
+                        mean = np.insert(mean, j, 0, axis=0)
+
+                cov_matrices = [measurement_model.covar()
+                                for _ in range(len(model.model_list))]
```

Comment:
Should this be transition model covariance (or transition model PDF method), rather than the measurement model?

Reply:
I've changed this to the transition model now.